### PR TITLE
MODFISTO-350 added python script with encumbrance relations fix

### PR DESCRIPTION
**Pull request by grabsefx**
## Purpose
* https://issues.folio.org/browse/MODFISTO-350

## Approach
* added python script with encumbrance relations fix

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: scripts/fix_lotus_encumbrances/fix_lotus_encumbrances.py [hunk 1]
```diff
@@ -496,22 +511,97 @@ async def release_unreleased_encumbrances_for_closed_orders(closed_orders_ids, f
...
+
+
+# Get po_lines by order id
+    # for each poline process fund distributions
+        # for each fund distribution check encumbrance relationship and modify if needed -
+        # in case if encumbrance id specified in fund distribution:
+        # get encumbrance by poline id and current FY<>transaction.FY and amount <> 0
+        # if fd.encumbrance != transaction.id --> set new encumbrance reference
+        # update poline if modified
+async def fix_poline_encumbrances_relations(open_orders_ids, fiscal_year_id, all_funds):
```
Comment (SerhiiNosko) [inline scripts/fix_lotus_encumbrances/fix_lotus_encumbrances.py:585]: please update readme with this new step:
https://github.com/folio-org/mod-finance-storage/blob/master/scripts/fix_lotus_encumbrances/README.md

**Review by Abdulkhakimov — APPROVED** [review]

File: scripts/fix_lotus_encumbrances/fix_lotus_encumbrances.py [hunk 2]
```diff
@@ -496,22 +511,97 @@ async def release_unreleased_encumbrances_for_closed_orders(closed_orders_ids, f
...
+        fd['encumbrance'] = encumbrances[0]['id']
+        return True
+    return False
+
+
+async def process_po_line_encumbrances_relations(poline, fiscal_year_id, all_funds):
+    enc_futures = []
+    for fd in poline['fundDistribution']:
+        if fd.get('encumbrance') is not None:
+            query = 'amount<>0.0 AND encumbrance.sourcePoLineId=={} AND fiscalYearId<>{} AND fromFundId=={}'.format(
```
Comment (NikitaSedyx) [inline scripts/fix_lotus_encumbrances/fix_lotus_encumbrances.py:567]: I believe query is incorrect
1. `encumbrance.sourcePoLineId=={}` should be line id, but `poline['purchaseOrderId']` is used, so it should be `poline['id']`
2. `fiscalYearId<>{}` should be `fiscalYearId=={}` as you try to find unlinked encumbrances in current FY, but <> means that in previous

in addition we should add a note, that this script can be executed only for current year, as we'll define links for previous years. Or we should check that FY is current one and skip step to fix references

File: scripts/fix_lotus_encumbrances/fix_lotus_encumbrances.py [hunk 3]
```diff
@@ -531,13 +621,15 @@ async def main():
...
+    all_funds = get_funds()
 
     # get open and closed order ids
-    closed_orders_ids = get_closed_orders_ids(fund_ids)
-    open_orders_ids = get_open_orders_ids(fund_ids)
+    closed_orders_ids = get_closed_orders_ids()
+    open_orders_ids = get_open_orders_ids()
     open_and_closed_orders_ids = closed_orders_ids + open_orders_ids
 
+    await fix_poline_encumbrances_relations(open_orders_ids, fiscal_year_id, all_funds)
```
Comment (NikitaSedyx) [inline scripts/fix_lotus_encumbrances/fix_lotus_encumbrances.py:631]: potentially (I can't say exactly) it's an issue to fix relations before we release encumbrances as some orders won't have links to unreleased encumbrances anymore

File: scripts/fix_lotus_encumbrances/fix_lotus_encumbrances.py (hunk 1 above)
Comment (grabsefx) [reply scripts/fix_lotus_encumbrances/fix_lotus_encumbrances.py:585]: updated readme file

**Review by SerhiiNosko — APPROVED** [review]

File: scripts/fix_lotus_encumbrances/fix_lotus_encumbrances.py (hunk 2 above)
Comment (grabsefx) [reply scripts/fix_lotus_encumbrances/fix_lotus_encumbrances.py]: fixed

**Pull request merged**